### DISCUSSION TITLE
[NEXT-792] dont add duplicate paths to tsconfig.json#include

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#65208] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023
    Binaries:
      Node: 18.14.0
      npm: 9.3.1
      Yarn: 1.22.19
      pnpm: 7.28.0
    Relevant packages:
      next: 13.2.4-canary.4
      eslint-config-next: 13.2.3
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

CLI (create-next-app)

### Link to the code that reproduces this issue

if necessary i will create one

### To Reproduce

- `pnpm create next-app . --experimental-app`
- edit `tsconfig.json` and change `".next/types/**/*.ts"` to `"./.next/types/**/*.ts"`
- `pnpm run dev`
- `cat tsconfig.json`, which has:

```json
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    "./.next/types/**/*.ts",
    ".next/types/**/*.ts"
  ],
```

### Describe the Bug

next.js automatically adds `.next/types/**/*.ts` to `tsconfig.json#include`.

however, it also does this when i already have `./.next/types/**/*.ts` in `tsconfig.json#include`. (note that this path starts with `./`)

### Expected Behavior

next.js should avoid adding duplicate entries to `tsconfig.json#include`

<sub>[NEXT-792](https://linear.app/vercel/issue/NEXT-792/dont-add-duplicate-paths-to-tsconfigjsoninclude)</sub>

**Comment by taep96** [comment]
Tried finding the commit that causes this but closest I could get is somewhere between 13.2.1 and 13.2.2

**Comment by AugustsK** [comment]
It seems that its the following change -> https://github.com/vercel/next.js/commit/b6b7933377e44137cb51a0674a79a4e8204f0c06

Previously `verifyTypeScriptSetup` was executed only on build. This commit adds it to `next lint` as well, but the `distDir` argument differs to what is passed during build.

**Comment by chungweileong94** [comment]
Yeah, it's somewhat annoying, especially when my codebase is in an external volume/drive, it adds absolute path instead every time I execute `next lint`.

**Comment by KubaJastrz** [comment]
>Yeah, it's somewhat annoying, especially when my codebase is in an external volume/drive, it adds absolute path instead every time I execute next lint.

Same thing happens in WSL2 environment.
